cvmfs/download.cc

```diff
@@ -231,9 +231,13 @@ static size_t CallbackCurlHeader(void *ptr, size_t size, size_t nmemb,
   if ((info->destination == kDestinationMem) &&
       HasPrefix(header_line, "CONTENT-LENGTH:", true))
   {
-    char *tmp = reinterpret_cast<char *>(alloca(num_bytes+1));
     uint64_t length = 0;
-    sscanf(header_line.c_str(), "%s %"PRIu64, tmp, &length);
+    size_t pos = header_line.find(":") + 1;
+    assert(header_line.length() > pos);
+    if (header_line[pos] == ' ') {
+      ++pos;
+    }
+    sscanf(&header_line[pos], "%"PRIu64, &length);
```

Comment:
Yes, just make sure that the header is long enough to contain at least one digit for the content length.  Then it should be fine.